components/UploadPopover/UploadPopover.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from "react";
+import styles from "./UploadPopover.module.css";
+import { Button, Popover, LinearProgress } from "@mui/material";
+import Icon from "../../assets/Icon";
+import { Transaction, User } from "../../types/schema";
+import { addTransaction } from "../../firebase/firestore/transaction";
+import Papa from '../../node_modules/papaparse';
+
+type UploadPopoverProps = {
+    uploadAnchor: Element;
+    closeUpload: Function;
+    popoverid: string;
+};
+
+export const UploadPopover: React.FunctionComponent<UploadPopoverProps> = ({
+  uploadAnchor,
+  closeUpload,
+  popoverid,
+}: UploadPopoverProps) => {
+    const [uploadFile, setUploadFile] = useState(null);
+    const [fileData, setFileData] = useState([]);
+    const [uploading, setUploading] = useState(false);
+    const [uploadProgress, setUploadProgress] = useState(0);
+    const [uploadSuccess, setUploadSuccess] = useState(false);
+
+    const sleep = (ms) => {
```

Comment:
if you get my suggested implementation of useeffect to work for your other PR, please make the same relevant changes here too!